pkg/jobstore/watcher.go

```diff
@@ -0,0 +1,114 @@
+package jobstore
+
+import "time"
+
+type StoreWatcherType int
+
+const (
+	JobWatcher StoreWatcherType = 1 << iota
+	ExecutionWatcher
+	EvaluationWatcher
+)
+
+func (s StoreWatcherType) String() string {
+	switch s {
+	case JobWatcher:
+		return "JobWatcher"
+	case ExecutionWatcher:
+		return "ExecutionWatcher"
+	case EvaluationWatcher:
+		return "EvaluationWatcher"
+	}
+	return "Unknown type"
+}
+
+type StoreEventType int
+
+const (
+	CreateEvent StoreEventType = 1 << iota
+	UpdateEvent
+	DeleteEvent
+)
+
+func (s StoreEventType) String() string {
+	switch s {
+	case CreateEvent:
+		return "CreateEvent"
+	case UpdateEvent:
+		return "UpdateEvent"
+	case DeleteEvent:
+		return "DeleteEvent"
+	}
+	return "Unknown event"
+}
+
+const DefaultWatchChannelSize = 64
+
+// WatchEvent is the message passed through the watcher whenever a
+// specific event occurs on a specific type, as requested when creating
+// the watcher.
+type WatchEvent struct {
+	Kind      StoreWatcherType
+	Event     StoreEventType
+	Object    []byte
+	Timestamp int64
+}
+
+func NewWatchEvent(kind StoreWatcherType, event StoreEventType, object []byte) WatchEvent {
+	return WatchEvent{
+		Kind:      kind,
+		Event:     event,
+		Object:    append([]byte(nil), object...),
+		Timestamp: time.Now().Unix(),
+	}
+}
+
+// Watcher is used by the jobstore to keep a record of parties interested in events happening
+// in the jobstore.  This allows for watching of job and execution types (or both), and for
+// create, update and delete events (or any combination).
+type Watcher struct {
+	types       StoreWatcherType // a bitmask of types being watched
+	events      StoreEventType   // a bitmask of events being watched
```

Comment:
a watcher is only watching a single object type and single event type, but the field names and comments say differently

Reply:
Watchers can mask the watcher type and the events (see https://github.com/bacalhau-project/bacalhau/pull/2650#discussion_r1275916813) 

So `store.JobWatcher | store.EvaluationWatcher` is valid, as is just `store.JobWatcher`.